Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/utils/FileBrowserHelper.java

```diff
@@ -85,4 +94,82 @@ public static String getSelectedPath(Intent result)
 
     return null;
   }
+
+  public static boolean isPathEmptyOrValid(StringSetting path)
+  {
+    return isPathEmptyOrValid(path.getStringGlobal());
+  }
+
+  public static boolean isPathEmptyOrValid(String path)
+  {
+    return !path.startsWith("content://") || ContentHandler.exists(path);
+  }
+
+  public static void runAfterExtensionCheck(Context context, Uri uri, Set<String> validExtensions,
+          Runnable runnable)
+  {
+    String extension = null;
+
+    String path = uri.getLastPathSegment();
+    if (path != null)
+      extension = getExtension(new File(path).getName());
+
+    if (extension == null)
+      extension = getExtension(ContentHandler.getDisplayName(uri));
+
+    if (extension != null && validExtensions.contains(extension))
+    {
+      runnable.run();
+      return;
+    }
+
+    String message;
+    if (extension == null)
+    {
+      message = context.getString(R.string.no_file_extension);
+    }
+    else
+    {
+      int messageId = validExtensions.size() == 1 ?
+              R.string.wrong_file_extension_single : R.string.wrong_file_extension_multiple;
+
+      ArrayList<String> extensionsList = new ArrayList<>(validExtensions);
+      Collections.sort(extensionsList);
+
+      message = context.getString(messageId, extension, join(", ", extensionsList));
+    }
+
+    new AlertDialog.Builder(context, R.style.DolphinDialogBase)
```

Comment:
Can this `AlertDialog` be not cancelable? This provides important information that something is likely incorrect.